src/dom-shim.js

```diff
@@ -45,10 +45,9 @@ class Element extends Node {
     return this.shadowRoot;
   }
 
-  // https://github.com/mfreed7/declarative-shadow-dom/blob/master/README.md#serialization
-  // eslint-disable-next-line
-  getInnerHTML() {
-    return this.shadowRoot ? this.shadowRoot.innerHTML : this.innerHTML;
+  getHTML({ serializableShadowRoots = false }) {
+    return this.shadowRoot && serializableShadowRoots ?
```

Comment:
So did some initial testing in Greenwood and observed one small caveat here, with this line in particular
```js
return this.shadowRoot && ...
```

In that technically you can `attachShadow` and return the value, for example
```js
export default class HeaderComponent extends HTMLElement {
  connectedCallback() {
    this.root = this.attachShadow({ mode: 'open' });
    this.root.innerHTML = `<header>This is the header component.</header>`;
  }
}

customElements.define('app-header', HeaderComponent);
```

Would work, at least as demonstrated by [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow#examples).

And so with this change we would assume users to _always_ have to do either one of the following it seems?
```js
// option 1
this.attachShadow({ mode: 'open' });
this.shadowRoot.innerHTML = `<header>This is the header component.</header>`;

// option 2
this.shadowRoot = this.attachShadow({ mode: 'open' });
this.shadowRoot.innerHTML = `<header>This is the header component.</header>`;
```

I think it's not that big of a deal, and if so I can make sure we cover this as part of #181 

@briangrider not sure if your PR would account for this, or should we just assume `this.shadowRoot`?

Reply:
Hm, I think I'm not understanding the issue here. I believe everything you mentioned (including this.root = this.attachShadow({ mode: 'open' });) should work as expected both before the DOM shim refactor and after, right? If this.attachShadow is called, this.shadowRoot will always exist and both DOM shims return this.shadowRoot when attaching shadow. But I think it's possible I'm not understanding the problem.

Reply:
Sorry yes, should have provided a bit more of the context but the test case that looked like it was failing was for this custom element definition
https://github.com/ProjectEvergreen/greenwood/blob/master/packages/cli/test/cases/build.config.prerender/src/components/header.js

In that I thought it was _not_ emitting the `<template>` tag
```html
<app-header>
  <header>This is the header component.</header>
</app-header>
```

But testing again after double checking my patching and it _does_ look the `<template>` tag is there
```html
<app-header>
  <template shadowrootmode="open">
    <header>This is the header component.</header>
  </template>
</app-header>
```

and so the test case just need to get updated to account for the (new) `<template>` which now appears and is the (new) expected behavior we would be seeing, yes? ([the selector](https://github.com/ProjectEvergreen/greenwood/blob/master/packages/cli/test/cases/build.config.prerender/build.config.prerender.spec.js#L103) just needs to account for addition child element of the `<template>`

So apologies, may have been a false alarm.  🙃

Reply:
Here's the draft PR for Greenwood for anyone who is curious, but I think we're all good now after I tweaked the test case to account for the new `<template>` tag.
https://github.com/ProjectEvergreen/greenwood/pull/1340

So I think with just the docs update here, we should be good to go to get this merged.

Reply:
According to the specs, `attachShadow` already returns `this.shadowRoot`.